src/ctap/mod.rs

```diff
@@ -307,20 +343,35 @@ where
                         Duration::from_ms(TOUCH_TIMEOUT_MS),
                     );
                 }
+                match (&command, &self.stateful_command_type) {
+                    (
+                        Command::AuthenticatorGetNextAssertion,
+                        Some(StatefulCommand::GetAssertion(_)),
+                    ) => (),
+                    (Command::AuthenticatorReset, Some(StatefulCommand::Reset)) => (),
+                    // GetInfo does not reset stateful commands.
+                    (Command::AuthenticatorGetInfo, _) => (),
+                    // AuthenticatorSelection does not reset stateful commands.
+                    #[cfg(feature = "with_ctap2_1")]
+                    (Command::AuthenticatorSelection, _) => (),
+                    (_, _) => {
+                        self.stateful_command_type = None;
+                    }
+                }
                 let response = match command {
                     Command::AuthenticatorMakeCredential(params) => {
                         self.process_make_credential(params, cid)
                     }
                     Command::AuthenticatorGetAssertion(params) => {
-                        self.process_get_assertion(params, cid)
+                        self.process_get_assertion(params, cid, now)
                     }
+                    Command::AuthenticatorGetNextAssertion => self.process_get_next_assertion(now),
                     Command::AuthenticatorGetInfo => self.process_get_info(),
                     Command::AuthenticatorClientPin(params) => self.process_client_pin(params),
-                    Command::AuthenticatorReset => self.process_reset(cid),
+                    Command::AuthenticatorReset => self.process_reset(cid, now),
                     #[cfg(feature = "with_ctap2_1")]
                     Command::AuthenticatorSelection => self.process_selection(cid),
-                    // TODO(kaczmarczyck) implement GetNextAssertion and FIDO 2.1 commands
-                    _ => self.process_unknown_command(),
```

Comment:
Why removing this?
If the command is unsupported we must reply.

Reply:
We do return an error_code in `Command::deserialize`. We needed `process_unknown_command` before when we correctly parsed the inputs of some commands, but didn't actually implement them. From now on, I don't expect that we will have a command that we parse, but don't handle here.

Reply:
I see. Then we're good.